docs/Style.thy

```diff
@@ -0,0 +1,182 @@
+(*
+ * Copyright 2021, Data61, CSIRO
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * @TAG(DATA61_BSD)
+ *)
+
+chapter \<open>An Isabelle Syntax Style Guide\<close>
+
+theory Style
+  imports Main
+begin
+
+(* This style guide is heavily influenced by Gerwin's Style Guide for Isabelle/HOL. See parts 1
+   and 2 here:
+
+   * https://proofcraft.org/blog/isabelle-style.html
+   * https://proofcraft.org/blog/isabelle-style-part2.html
+
+   This guide may repeat some parts of Gerwin's guide and should not conflict in any way.
```

Comment:
There is also a difference between the style guides: ny style guide avoids almost all syntactic considerations and is mostly about semantics things. This one here is more focussed on the syntactic aspect. Ideally we'd just link to mine for semantic considerations (like "auto in the middle of a proof") or when we have a different rule.